spec.md

```diff
@@ -0,0 +1,547 @@
+# Container Object Storage Interface (COSI)
+
+Authors:
+
+* Sidharth Mani  << @wlan0
+* Srini Brahmaroutu <<srbrahma@us.ibm.com>> (@brahmaroutu)
+
+## Notational Conventions
+
+The keywords "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+
+The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard](http://www.open-std.org/jtc1/sc22/wg14/www/C99RationaleV5.10.pdf#page=18).
+
+An implementation is not compliant if it fails to satisfy one or more of the MUST, REQUIRED, or SHALL requirements for the protocols it implements.
+An implementation is compliant if it satisfies all the MUST, REQUIRED, and SHALL requirements for the protocols it implements.
+
+## Terminology
+
+| Term              | Definition                                       |
+|-------------------|--------------------------------------------------|
+| Bucket            | A flat organization of container that store files.                          |
+| CO                | Container Orchestration system, communicates with Plugins using COSI service RPCs.                                     |
+| SP                | Storage Provider, the vendor of a COSI plugin implementation.                                                          |
+| RPC               | [Remote Procedure Call](https://en.wikipedia.org/wiki/Remote_procedure_call).                                         |
+| Node              | A host where the user workload will be running, uniquely identifiable from the perspective of a Plugin by a node ID. |
+| Plugin            | Aka “plugin implementation”, a gRPC endpoint that implements the COSI Services.                                        |
+| Plugin Supervisor | Process that governs the lifecycle of a Plugin, MAY be the CO.                                                        |
+| Workload          | The atomic unit of "work" scheduled by a CO. This MAY be a container or a collection of containers.                   |
+
+## Objective
```

Comment:
This _feels_ like a separate, stand-alone KEP? Is that the intention?

Reply:
I felt that way too,  but the spec repo should have a reference document. KEP we have belongs to K8s as a gating mechanism to get cost features in. 
Once a COSI repo is stood up, it will contain this document as well as 'cosi.proto' just like a CSI repo.